File: common/src/main/java/com/ultreon/mods/advanceddebug/client/input/KeyBindingList.java
package com.ultreon.mods.advanceddebug.client.input;

import com.mojang.blaze3d.platform.InputConstants;
import dev.architectury.registry.client.keymappings.KeyMappingRegistry;
import net.minecraft.client.KeyMapping;

import java.util.ArrayList;
import java.util.List;

/**
 * Keybinding list class.
 *
 * @author XyperCode
 */
public class KeyBindingList {
    public static final List<KeyMapping> KEY_BINDINGS = new ArrayList<>();
    public static final KeyMapping DEBUG_SCREEN = add(new KeyMapping("key.advanced_debug.debug_screen", InputConstants.KEY_F4, "key.categories.advanced_debug"));
    public static final KeyMapping SELECT_ENTITY = add(new KeyMapping("key.advanced_debug.selectEntity", InputConstants.KEY_F9, "key.categories.advanced_debug"));
    public static final KeyMapping SELECT_BLOCK = add(new KeyMapping("key.advanced_debug.selectBlock", InputConstants.KEY_F10, "key.categories.advanced_debug"));

    public static void register() {
        // Actually register all keys
        for (KeyMapping keyBinding : KEY_BINDINGS) {
            KeyMappingRegistry.register(keyBinding);
        }
    }

    private static KeyMapping add(KeyMapping keyBinding) {
        KEY_BINDINGS.add(keyBinding);
        return keyBinding;
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/client/input/KeyBindingList.kt
package com.ultreon.mods.advanceddebug.client.input

import dev.architectury.registry.client.keymappings.KeyMappingRegistry
import net.minecraft.client.KeyMapping

/**
 * Keybinding list class.
 *
 * @author Qboi123
 */
object KeyBindingList {
    private val KEY_BINDINGS: MutableList<KeyMapping> = ArrayList()
    @JvmField
    var DEBUG_SCREEN = add(KeyMapping("key.advanced_debug.debug_screen", 293, "key.categories.advanced_debug"))
    @JvmStatic
    fun register() {
        // Actually register all keys
        for (keyBinding in KEY_BINDINGS) {
            KeyMappingRegistry.register(keyBinding)
        }
    }

    private fun add(keyBinding: KeyMapping): KeyMapping {
        KEY_BINDINGS.add(keyBinding)
        return keyBinding
    }
}